trac_ik: 1.4.9-0 in 'lunar/distribution.yaml' [bloom]

pbeeson opened the pull request:
Increasing version of package(s) in repository `trac_ik` to `1.4.9-0`:

- upstream repository: https://bitbucket.org/traclabs/trac_ik.git
- release repository: https://github.com/traclabs/trac_ik-release.git
- distro file: `lunar/distribution.yaml`
- bloom version: `0.6.1`
- previous version for package: `1.4.8-0`

## trac_ik

```
* Fixed MoveIt! plugin params to look in the correct place
* Added a swig wrapper around TRAC-IK courtesy of mailto:Sammy.Pfeiffer@student.uts.edu.au
```

mikaelarguedas commented:
@pbeeson @awesomebytes FYI: The SWIG wrapper library in trac_ik_python fails to build on all platforms on the ROS buildfarm. (e.g. http://build.ros.org/job/Lbin_uX64__trac_ik_python__ubuntu_xenial_amd64__binary/)

pbeeson commented:
1) This builds for me in all my prerelease attempts.
2) Why am I not getting emails about this jenkins failure?

pbeeson commented:
Someting weird is going on here:
```
/usr/lib/ccache/x86_64-linux-gnu-g++  -fPIC -std=c++0x -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -shared  -o devel/lib/python2.7/dist-packages/trac_ik_python/_trac_ik_wrap.so CMakeFiles/_trac_ik_wrap.dir/trac_ik_wrapPYTHON_wrap.c.o -ltrac_ik -lpython2.7 /opt/ros/lunar/lib/libtrac_ik.so /opt/ros/lunar/lib/libkdl_parser.so /opt/ros/lunar/lib/liburdf.so -lurdfdom_sensor -lurdfdom_model_state -lurdfdom_model -lurdfdom_world -ltinyxml /opt/ros/lunar/lib/librosconsole_bridge.so /opt/ros/lunar/lib/libtf_conversions.so /opt/ros/lunar/lib/libkdl_conversions.so /opt/ros/lunar/lib/liborocos-kdl.so.1.3.0 /opt/ros/lunar/lib/libtf.so /opt/ros/lunar/lib/libtf2_ros.so /opt/ros/lunar/lib/libactionlib.so /opt/ros/lunar/lib/libmessage_filters.so /opt/ros/lunar/lib/libroscpp.so -lboost_filesystem -lboost_signals /opt/ros/lunar/lib/libxmlrpcpp.so /opt/ros/lunar/lib/libtf2.so /opt/ros/lunar/lib/libroscpp_serialization.so /opt/ros/lunar/lib/librosconsole.so /opt/ros/lunar/lib/librosconsole_log4cxx.so /opt/ros/lunar/lib/librosconsole_backend_interface.so -llog4cxx -lboost_regex /opt/ros/lunar/lib/librostime.so /opt/ros/lunar/lib/libcpp_common.so -lboost_system -lboost_thread -lboost_chrono -lboost_date_time -lboost_atomic -lpthread -lconsole_bridge -Wl,-rpath,/opt/ros/lunar/lib: 
23:14:32 /usr/bin/ld: cannot find -ltrac_ik
```

This implies trac_ik is explcitly set on the LINK_LIBRARIES line.  I believe this was the case when @awesomebytes first submitted his PR, but I fixed this and its not in the 1.4.9 tag that is supposed to be getting pulled.

I did update the tag to a different commit blob after running bloom-release, might this be this issue?  Is there something in release or the like that binds 1.4.9 to a specific commit?

mikaelarguedas commented:
> Why am I not getting emails about this jenkins failure?

Looks like you're not listed as a maintainer of this package, if you add your name to the maintainer list you will receive emails: https://bitbucket.org/traclabs/trac_ik/src/660c7cbae97c5960ecb772b477a4c31818a0daed/trac_ik_python/package.xml?at=master&fileviewer=file-view-default#package.xml-8 

> but I fixed this after release, and updated the 1.4.9 to the new commit.

the version on the release repository doesnt have the fix: https://github.com/traclabs/trac_ik-release/blob/721d57f79c507756f55299b312f22543206248f3/CMakeLists.txt#L68

Did you rerun bloom after updating the tag ?
When being ran, bloom imports the code at the latest tag and commits it to the release repository. So it is necessary to rerun bloom if any modification is made to the tags on the source repository.
As changing tags may have undefined/unwanted side effects in the release process, we recommend releasing a new version for any code / cmake change rather than moving the tag to other commits.

Hope this helps

pbeeson commented:
I assumed the release repo was scythed to a tag not a commit blob. This
explains it.
On Sat, Jan 20, 2018 at 1:46 PM Mikael Arguedas <notifications@github.com>
wrote:

> Why am I not getting emails about this jenkins failure?
>
> Looks like you're not listed as a maintainer of this package, if you add
> your name to the maintainer list you will receive emails:
> https://bitbucket.org/traclabs/trac_ik/src/660c7cbae97c5960ecb772b477a4c31818a0daed/trac_ik_python/package.xml?at=master&fileviewer=file-view-default#package.xml-8
>
> but I fixed this after release, and updated the 1.4.9 to the new commit.
>
> the version on the release repository doesnt have the fix:
> https://github.com/traclabs/trac_ik-release/blob/721d57f79c507756f55299b312f22543206248f3/CMakeLists.txt#L68
>
> Did you rerun bloom after updating the tag ?
> When being ran, bloom imports the code at the latest tag and commits it to
> the release repository. So it is necessary to rerun bloom if any
> modification is made to the tags on the source repository.
> As changing tags may have undefined/unwanted side effects in the release
> process, we recommend releasing a new version for any code / cmake change
> rather than moving the tag to other commits.
>
> Hope this helps
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ros/rosdistro/pull/16761#issuecomment-359197171>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADrfxvlYj30N3_mVZnv9KMkYsspR9DjBks5tMkKXgaJpZM4RbmOT>
> .
>

mikaelarguedas commented:
The release repository is indeed synced to a tag, but the sync is performed only when bloom-release is ran. This allows to keep building binaries even if the source tags or repo get compromised.
